src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpRename.cs

```diff
@@ -81,6 +81,99 @@ static void TestMethod(int y)
             }
         }
 
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Rename)]
+        [WorkItem(21657, "https://github.com/dotnet/roslyn/issues/21657")]
+        public void VerifyAttributeRename()
+        {
+            var markup = @"
+using System;
+
+class [|$$ustom|]Attribute : Attribute
+{
+}
+";
+            SetUpEditor(markup);
+            InlineRenameDialog.Invoke();
+
+            MarkupTestFile.GetSpans(markup, out var _, out ImmutableArray<TextSpan> renameSpans);
+            var tags = VisualStudio.Editor.GetTagSpans(InlineRenameDialog.ValidRenameTag);
+            AssertEx.SetEqual(renameSpans, tags);
+
+            VisualStudio.Editor.SendKeys("Custom", VirtualKey.Enter);
+            VisualStudio.Editor.Verify.TextContains(@"
+using System;
+
+class CustomAttribute : Attribute
+{
+}");
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Rename)]
+        [WorkItem(21657, "https://github.com/dotnet/roslyn/issues/21657")]
+        public void VerifyAttributeRenameWhileRenameClasss()
+        {
+            var markup = @"
+using System;
+
+class [|$$stom|]Attribute : Attribute
+{
+}
+";
+            SetUpEditor(markup);
+            InlineRenameDialog.Invoke();
+
+            MarkupTestFile.GetSpans(markup, out var _, out ImmutableArray<TextSpan> renameSpans);
+            var tags = VisualStudio.Editor.GetTagSpans(InlineRenameDialog.ValidRenameTag);
+            AssertEx.SetEqual(renameSpans, tags);
+
+            VisualStudio.Editor.SendKeys("Custom");
+            VisualStudio.Editor.Verify.TextContains(@"
+using System;
+
+class Custom$$Attribute : Attribute
+{
+}
+",true);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Rename)]
+        [WorkItem(21657, "https://github.com/dotnet/roslyn/issues/21657")]
+        public void VerifyAttributeRenameWhileRenameAttribute()
+        {
+            var markup = @"
+using System;
+
+[[|$$stom|]]
```

Comment:
Do we have a test with same source that does identical rename but on the definition and verifies the attribute applications are fixed, i.e.:

Source:
```
using System;

[stom]
class Bar 
{
}

class [[|$$stom|]]Attribute : Attribute
{
}
```

Result:
```
using System;

[Custom]
class Bar 
{
}

class [Custom$$]Attribute : Attribute
{
}
```

Reply:
I add this test at:

```csharp
VerifyAttributeRenameWhileRenameAttributeClass
```